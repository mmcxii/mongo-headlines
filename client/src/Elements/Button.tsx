import styled from 'styled-components';
import { Link } from 'react-router-dom';

import { rounded, fallingStar, blueBell, clearChill, spacing } from 'Utilities';

export const Button = styled.button`
    cursor: pointer;
    text-align: center;
    border-radius: ${rounded};
    border: none;
    padding: ${spacing.sm} ${spacing.md};
    margin: ${spacing.xs};
    font-weight: bolder;
    color: ${fallingStar};
    background: ${blueBell};
    transition: background ease-in 250ms;

    &:hover {
        background: ${clearChill};
    }
`;

export const ButtonLink = styled(Link)`
    cursor: pointer;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    border-radius: ${rounded};
    border: none;
    padding: ${spacing.sm} ${spacing.md};
    margin: ${spacing.xs};
    font-weight: bolder;
    color: ${fallingStar};
    background: ${blueBell};
    transition: background ease-in 250ms;

    &:hover {
        background: ${clearChill};
    }
`;
