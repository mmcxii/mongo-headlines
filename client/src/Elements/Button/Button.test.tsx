import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import { Button, ButtonLink } from './Button';

describe('<Button /> tests', () => {
    it('renders a button with the text `hello`', () => {
        //* Arrange
        let greeting = 'hello';
        const sayHi = () => (greeting = 'hi');

        //* Act
        const { container, getByTestId, rerender } = render(
            <Button data-testid='testButton' onClick={sayHi}>
                {greeting}
            </Button>
        );
        const testButton = container.firstChild;

        //* Assert
        expect(testButton).toMatchSnapshot();
        fireEvent.click(getByTestId('testButton'));
        rerender(
            <Button data-testid='testButton' onClick={sayHi}>
                {greeting}
            </Button>
        );
        expect(getByTestId('testButton').textContent).toBe('hi');
    });
});

describe('<ButtonLink /> tests', () => {
    it('renders a link to the homepage with button stylings', () => {
        //* Act
        const { container } = render(
            <Router>
                <ButtonLink to='/'>Home</ButtonLink>
            </Router>
        );

        //* Assert
        expect(container.firstChild).toMatchSnapshot();
    });
});
