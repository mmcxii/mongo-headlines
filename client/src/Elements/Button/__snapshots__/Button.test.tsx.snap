// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> tests renders a button with the text \`hello\` 1`] = `
.c0 {
  cursor: pointer;
  text-align: center;
  border-radius: 5px;
  border: none;
  padding: .5rem 1rem;
  margin: .25rem;
  font-weight: bolder;
  color: #CAD3C8;
  background: #3B3B98;
  -webkit-transition: background 300ms ease;
  transition: background 300ms ease;
}

.c0:hover {
  background: #1B9CFC;
}

<button
  class="c0"
  data-testid="testButton"
>
  hello
</button>
`;

exports[`<ButtonLink /> tests renders a link to the homepage with button stylings 1`] = `
.c0 {
  cursor: pointer;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  border-radius: 5px;
  border: none;
  padding: .5rem 1rem;
  margin: .25rem;
  font-weight: bolder;
  color: #CAD3C8;
  background: #3B3B98;
  -webkit-transition: background 300ms ease;
  transition: background 300ms ease;
}

.c0:hover {
  background: #1B9CFC;
}

<a
  class="c0"
  href="/"
>
  Home
</a>
`;
